src/jit/codegenxarch.cpp

```diff
@@ -6620,26 +6611,6 @@ void CodeGen::genIntToIntCast(GenTreePtr treeNode)
         {
             ins = INS_mov;
         }
-        /* Is the value sitting in a non-byte-addressable register? */
-        else if (castOp->InReg() && (dstSize == EA_1BYTE) && !isByteReg(sourceReg))
-        {
-            if (isUnsignedDst)
-            {
-                // for unsigned values we can AND, so it need not be a byte register
-                ins = INS_AND;
-            }
-            else
-            {
-                // Move the value into a byte register
-                noway_assert(!"Signed byte convert from non-byte-addressable register");
-            }
-
-            /* Generate "mov targetReg, castOp->gtReg */
-            if (targetReg != sourceReg)
-            {
-                inst_RV_RV(INS_mov, targetReg, sourceReg, srcType, srcSize);
-            }
-        }
 
```

Comment:
This code was dead, since `InReg()` always returned zero. Doing anything else but deleting this code would change behavior, which I was specifically avoiding.